src/main/java/it/gov/pagopa/reward/connector/repository/HpanInitiativesAtomicOpsRepositoryImpl.java

```diff
@@ -72,4 +74,18 @@ public Mono<UpdateResult> setInitiative(String hpan, OnboardedInitiative onboard
                     });
         }
     }
+
+    @Override
+    public Mono<UpdateResult> setIfNotEqualsStatus(String userId, String initiativeId, HpanInitiativeStatus status) {
+        return mongoTemplate
+                .updateMulti(
+                        Query.query(Criteria.where(HpanInitiatives.Fields.userId).is(userId)
+                                .and(FIELD_INITIATIVE_ID).is(initiativeId)
+                                .and(FIELD_STATUS).ne(status)),
+                        new Update()
+                                .set("%s.$.%s".formatted(HpanInitiatives.Fields.onboardedInitiatives, OnboardedInitiative.Fields.status), status)
+                                .currentDate("%s.$.%s".formatted(HpanInitiatives.Fields.onboardedInitiatives, OnboardedInitiative.Fields.updateDate)),
```

Comment:
store as constant field names instead of let the applicatication to resolve the formatted at each invocation